File: chapter8_library.r
print_stars <- function( list ){
  for( item in list ) {
    item = as.integer( item )
    print( paste( rep( '*', item ) , collapse = "" ) )
  }
}

mega_collector <- list()

data <- read.csv('experimental_data.txt')

for( i in 1:nrow(data) ){
  condition <- data[i, 1]
  measurement <- as.numeric( data[i, 2] )

  if( ! condition %in% names( mega_collector ) ){
    mega_collector[[ condition ]] <- list()
  }

  previous_values <- mega_collector[[ condition ]]
  new_values <- c( previous_values , measurement  )
  mega_collector[[ condition ]] <- new_values

}

for( condition in names(mega_collector) ) {
  measurements <- mega_collector[[ condition ]]
  measurements <- unlist( measurements ) ## transforming them to a simpler list
  m <- mean( measurements )
  v <- var( measurements )
  l <- c( m, v )
  print_stars( l )
}

File: chapter8_library.py
import statistics

def visualise( list ):
  for item in list:
    item = int( item )
    print( item * '*' )

mega_collector = {}

for line in open('experimental_data.txt'):
    condition = line.split(',')[0].strip()
    measurement = float( line.split(',')[1].strip() )

    if condition not in mega_collector:
      mega_collector[ condition ] = []

    mega_collector[ condition ].append( measurement )

for condition, measurements in mega_collector.items():
  m = statistics.mean( measurements )
  v = statistics.variance( measurements )
  items = []
  items.append( m )
  items.append( v )
  visualise( items )
